sdk/bpf/rust-utils/src/allocator.rs

```diff
@@ -0,0 +1,23 @@
+//! @brief Solana Rust-based BPF program memory allocator shim
+
+use crate::log::*;
+use core::alloc::{GlobalAlloc, Layout};
+
+pub struct Allocator;
+unsafe impl GlobalAlloc for Allocator {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        sol_alloc_free_(layout.size() as u64, 0)
```

Comment:
why the _ at the end of sol_alloc_free?  Just curious

Reply:
To separate the internal namespace from the external, allows us export a C and Rust version of `sol_log` who both call the same C function `sol_log_`.  Similar to the LLVM `builtin` namespaces.